### PR TITLE
Add pnet_tests

### PR DESCRIPTION
**Pull request by miri64**
Test driver for RIOT-OS/RIOT#226

### CODE REVIEW

**Comment by mehlis** [comment]
my msba2 is continuously restarting:

```
2013-12-23 14:12:27,989 - INFO # Board initialized.
2013-12-23 14:12:27,994 - INFO # kernel_init(): This is RIOT! (Version: 2013.08-823-geb63-pnet)
2013-12-23 14:12:27,995 - INFO # Scheduler...[OK]
2013-12-23 14:12:27,999 - INFO # kernel_init(): jumping into first task...
2013-12-23 14:12:28,001 - INFO # UART0 thread started.
2013-12-23 14:12:28,003 - INFO # uart0_init() [OK]
2013-12-23 14:12:28,005 - INFO # Unknown transceiver
2013-12-23 14:12:28,012 - INFO # Unknown transceiver
```

**Comment by mehlis** [comment]
building for native reveals a lot of problems. I used `BOARD=native LANG=C CC=clang CFLAGS=" -DHAVE_VALGRIND_VALGRIND_H -O0 -g -W -Wextra -pedantic" make -B clean all`

**Comment by miri64** [comment]
Fixed errors on my part.

```
/usr/include/x86_64-linux-gnu/bits/sockaddr.h:28:28: error: typedef redefinition with different types ('unsigned short' vs
      'uint8_t' (aka 'unsigned char'))
typedef unsigned short int sa_family_t;
                           ^
```

and the errors in `native-uart0.h` are not mine ;-)

**Comment by miri64** [comment]
@mehlis my msba2 too. I'm looking into it.

**Comment by LudwigKnuepfer** [comment]
What's that with errors in native-uart?

**Comment by mehlis** [comment]
currently there is no warning on the native part on my machine (you have to cherry-pick https://github.com/RIOT-OS/RIOT/pull/468)

**Comment by miri64** [comment]
@LudwigOrtmann after the latest patch they do not show anymore… RIOT-OS/RIOT#468 might have fixed it
@mehlis msba2 runs now. I forgot to include the cc110x_ng module.

**Comment by LudwigKnuepfer** [comment]
https://github.com/RIOT-OS/RIOT/pull/468 has been merged..

File: test_pnet/main.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+int main(void) {
+    int sockfd, res;
+    struct sockaddr_in6 my_addr, their_addr = {
+        .sin6_family = AF_INET6,
+        .sin6_port = 0,
+        .sin6_flowinfo = 0,
+#if R_ADDR == 1
+        .sin6_addr = {{0xab, 0xcd, 0xef, 0x12, 0, 0, 0x10, 0x34, 0, 0xff, 0xfe,
```
Comment (OlegHahm) [inline test_pnet/main.c]: Address is two short (only 14 bytes instead of 16).

File: test_pnet/main.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                       0, 0, 2}},
+#else
+        .sin6_addr = {{0xab, 0xcd, 0xef, 0x12, 0, 0, 0x10, 0x34, 0, 0xff, 0xfe,
```
Comment (OlegHahm) [inline test_pnet/main.c]: Address is two short (only 14 bytes instead of 16).

File: test_pnet/main.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    ipv6_addr_init(&std_addr, 0xabcd, 0xef12, 0, 0, 0x1034, 0x00ff, 0xfe00, 
+                   r_addr);
+    sixlowpan_lowpan_adhoc_init(TRANSCEIVER, &std_addr, r_addr);
+}
+
+int main(void) {
+    int sockfd, res;
+    struct sockaddr_in6 my_addr, their_addr = {
+        .sin6_family = AF_INET6,
+        .sin6_port = 0,
```
Comment (OlegHahm) [inline test_pnet/main.c:36]: Should be PORT.

File: test_pnet/main.c [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+#if R_ADDR == 1
+        .sin6_addr = {{0xab, 0xcd, 0xef, 0x12, 0, 0, 0x10, 0x34, 0, 0xff, 0xfe,
+                       0, 0, 2}},
+#else
+        .sin6_addr = {{0xab, 0xcd, 0xef, 0x12, 0, 0, 0x10, 0x34, 0, 0xff, 0xfe,
+                       0, 0, 1}},
+#endif
+        .sin6_scope_id = 0,
+    };
+    char *buffer = "Hello, world!";
```
Comment (OlegHahm) [inline test_pnet/main.c]: Doesn't this create a constant (read-only) char array? Segfaults and data aborts in native and MSB-A2 seem to support this theory.

File: test_pnet/main.c [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    init_local_address(R_ADDR);
+
+    memcpy(&my_addr, &in6addr_any, sizeof (my_addr));
```
Comment (OlegHahm) [inline test_pnet/main.c:52]: This leads to

```
Wrong sender address: 255 
```

Comment (miri64) [inline test_pnet/main.c:52]: Definitely not what it should do. Or am I mistaken?

**Comment by miri64** [comment]
Consolidated to RIOT-OS/RIOT#226

**Pull request closed without merging**